### ISSUE TITLE
[Validator] Choice constraint callback does not accepts array anymore

### ISSUE THREAD

**Issue by symfonyaml** [symfony/symfony-docs#19313]
Hi,

_Forgive me if I'm wrong._
- **Docs** : The documentation about the [Choice#callback](https://symfony.com/doc/current/reference/constraints/Choice.html#callback) constraint says that the available types are `string|array|Closure`.
- **Code** : The [Choice](https://github.com/symfony/symfony/blob/7.0/src/Symfony/Component/Validator/Constraints/Choice.php#L53) class constructor only accepts `callable|string` now since Symfony version `6.x`.

Basically the doc says we can do this :
```php
#[Assert\Choice(callback: [Genre::class, 'getGenres'])]
protected string $genre;
```
which throws the following error
```
TypeError: Symfony\Component\Validator\Constraints\Choice::__construct():
Argument #3 ($callback) must be of type callable|string|null, array given
```

**Comment by xabbuh** [comment]
Can you create a small example application that allows to reproduce your issue?

**Comment by xabbuh** [comment]
By the way, in your example `getGenres()` must be a public static method of the `Genre` class. Is that the case?

**Comment by symfonyaml** [comment]
> By the way, in your example `getGenres()` must be a public static method of the `Genre` class. Is that the case?

The example is just a copy of the docs in the [Supplying the Choices with a Callback Function](https://symfony.com/doc/current/reference/constraints/Choice.html#supplying-the-choices-with-a-callback-function) section.
Ok I will try to create a small example application.

**Comment by alexandre-daubois** [comment]
> By the way, in your example `getGenres()` must be a public static method of the `Genre` class. Is that the case?

I think this is it. I tried myself and it worked well on 6.4 with PHP 8.2. When removing either static or public, I have the same error. However, the doc needs an update indeed. I'll provide it :+1:

**Comment by symfonyaml** [comment]
I confirm I had this error when removing either static or public.
